src/connectors/dynamic-widgets/connectDynamicWidgets.ts

```diff
@@ -180,6 +229,17 @@ const connectDynamicWidgets: DynamicWidgetsConnector =
             );
           }
 
+          warning(
+            maxValuesPerFacet >= (state.maxValuesPerFacet || 0),
+            `The maxValuesPerFacet set by dynamic widgets (${maxValuesPerFacet}) is smaller than one of the limits set by a widget (${state.maxValuesPerFacet}). This causes a mismatch in query parameters and thus an extra network request when that widget is mounted.`
```

Comment:
I think this message could be helpful if it:
- stated the name of the widget -> I know it probably adds a lot more complexity.
- said something like: look for any child widget (`menu` , `hierarhicalMenu` or `refinementList`) using `limit` and `showMoreLimit` and make sure to keep `maxValuesPerFacet` superior.